### PR TITLE
New way to Handle the errors.

### PR DESCRIPTION
**Pull request by alejandroGonGon**
### Changes in the code:
Added the FormRequest to the instances where the Validation is used to handle the errors, being more indiviudal and easy to change and add rules to the attributes.

### Attachments
image

### Documentation
[Form Request Laravel](https://laravel.com/docs/5.7/validation#form-request-validation)

### CODE REVIEW

File: app/Http/Requests/PetitionStoreFormRequest.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            'subject' => 'required',
+            'description' => 'required',
+        ];
+    }
+    public function messages()
+    {
+        return [
+            // 'message' => 'Algo ha salido mal',
+            'subject.required' => 'El campo del Titulo es requerido',
+            'description.required' => 'La descripcion es requerida',
```
Comment (rubenderigo) [inline app/Http/Requests/PetitionStoreFormRequest.php:36]:
```suggestion
            'description.required' => 'La descripción es requerida',
```

File: app/Http/Requests/StoreDonationFormRequest.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            'image' => 'required|mimes:jpeg,bmp,png|max:1000',
+        ];
+    }
+
+    public function messages()
+    {
+        return [
+            // 'message' => 'Algo ha salido mal',
+            'name.required' => 'El campo de Titulo es requerido',
+            'description.required' => 'El campo de Descripcion es requerido',
```
Comment (rubenderigo) [inline app/Http/Requests/StoreDonationFormRequest.php:38]:
```suggestion
            'description.required' => 'El campo de Descripción es requerido',
```

File: app/Http/Requests/StoreDonationFormRequest.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            'image.required' => 'Una imagen es requerida',
+            'image.image' => 'Se necesita una Imagen valida',
+            'image.max' => 'El maximo de la imagen es de 5Mb',
```
Comment (rubenderigo) [inline app/Http/Requests/StoreDonationFormRequest.php:41]:
```suggestion
            'image.max' => 'El máximo de la imagen es de 5Mb',
```

File: app/Http/Requests/UpdateFormRequest.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    public function messages()
+    {
+        return [
+            // 'message' => 'Algo ha salido mal',
+            'email.required' => 'El campo Email es requerido',
+            'password.required' => 'El campo es requerido',
+            'name.required' => 'El campo Nombre es requerido',
+            'name.string' => 'Indique un nombre valido',
+            'phone.numeric' => 'Porfavor indique un numero',
```
Comment (rubenderigo) [inline app/Http/Requests/UpdateFormRequest.php:42]:
```suggestion
            'phone.numeric' => 'Por favor indique un número',
```

File: app/Http/Requests/UpdateFormRequest.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            'about_me.string' => 'Porfavor indique solo texto',
```
Comment (rubenderigo) [inline app/Http/Requests/UpdateFormRequest.php:43]:
```suggestion
            'about_me.string' => 'Por favor indique solo texto',
```

File: app/Http/Requests/UpdateFormRequest.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            'image.image' => 'No se pudo subir la imagen',
+            'image.max' => 'El maximo de la imagen es de 1Mb',
```
Comment (rubenderigo) [inline app/Http/Requests/UpdateFormRequest.php:45]:
```suggestion
            'image.max' => 'El máximo de la imagen es de 1Mb',
```

**Review by rubenderigo — APPROVED** [review]

File: app/Http/Requests/DonationRequestFormRequest.php [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'reason' => 'required|string',
+        ];
+    }
+    public function messages()
```
Comment (nicocadq) [inline app/Http/Requests/DonationRequestFormRequest.php:30]:
```suggestion
    }
    
    public function messages()
```

File: app/Http/Requests/DonationRequestFormRequest.php [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        return [
+            // 'message' => 'Algo ha salido mal',
```
Comment (nicocadq) [inline app/Http/Requests/DonationRequestFormRequest.php:33]:
```suggestion
```

File: app/Http/Requests/LogInFormRequest.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        return [
+            'email' => 'required|email:rfc,dns',
+            'password' => 'required|string'
+        ];
+    }
+    
+    public function messages()
+    {
+        return [
+            // 'message' => 'Algo ha salido mal',
```
Comment (nicocadq) [inline app/Http/Requests/LogInFormRequest.php:35]:
```suggestion
```

File: app/Http/Requests/LogInFormRequest.php [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+            'email.required' => 'Este campo es requerido',
+            'password.required' => 'El campo es requerido',
```
Comment (nicocadq) [inline app/Http/Requests/LogInFormRequest.php:37]: Try to be consistent
```suggestion
            'email.required' => 'Este campo es requerido',
            'password.required' => 'Este campo es requerido',
```

File: app/Http/Requests/PetitionStoreFormRequest.php [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        return [
+            'subject' => 'required',
+            'description' => 'required',
+        ];
+    }
+    public function messages()
+    {
+        return [
+            // 'message' => 'Algo ha salido mal',
+            'subject.required' => 'El campo del Titulo es requerido',
```
Comment (nicocadq) [inline app/Http/Requests/PetitionStoreFormRequest.php:35]: Same here

```suggestion
    }
    
    public function messages()
    {
        return [
            'subject.required' => 'El Titulo es requerido',
```

File: app/Http/Requests/RegisterFormRequest.php [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    public function messages()
+    {
+        return [
+            // 'message' => 'Algo ha salido mal',
+            'email.unique' => 'Este Email ya existe',
+            'email.required' => 'El campo Email es requerido',
+            'email.email' => 'Ingrese un Email valido',
+            'name.required' => 'El campo es requerido',
+            'password.required' => 'El campo es requerido',
+        ];
```
Comment (nicocadq) [inline app/Http/Requests/RegisterFormRequest.php:46]:
```suggestion
            'email.unique' => 'Este email ya existe',
            'email.required' => 'El email es requerido',
            'email.email' => 'Ingrese un email valido',
            'name.required' => 'El nombre es requerido',
            'password.required' => 'La contraseña es requerida',
        ];
```

File: app/Http/Requests/StoreDonationFormRequest.php [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+            'name' => 'required|string',
+            'description' => 'required|string',
+            'image' => 'required|mimes:jpeg,bmp,png|max:1000',
+        ];
+    }
+
+    public function messages()
+    {
+        return [
+            // 'message' => 'Algo ha salido mal',
```
Comment (nicocadq) [inline app/Http/Requests/StoreDonationFormRequest.php:36]:
```suggestion
```

File: app/Http/Requests/StoreDonationFormRequest.php (hunk 2 above)
Comment (nicocadq) [inline app/Http/Requests/StoreDonationFormRequest.php:38]:
```suggestion
            'name.required' => 'El Titulo es requerido',
            'description.required' => 'La Descripcion es requerida',
```

File: app/Http/Requests/StoreDonationFormRequest.php (hunk 3 above)
Comment (nicocadq) [inline app/Http/Requests/StoreDonationFormRequest.php:41]:
```suggestion
            'image.max' => 'El tamaño maximo de la imagen puede ser de hasta 5Mb',
```

File: app/Http/Requests/UpdateFormRequest.php (hunk 4 above)
Comment (nicocadq) [inline app/Http/Requests/UpdateFormRequest.php:42]:
```suggestion
            'email.required' => 'El email es requerido',
            'password.required' => 'La contraseña es requerida',
            'name.required' => 'El nombre es requerido',
            'name.string' => 'Indique un nombre valido',
            'phone.numeric' => 'Porfavor indique un número',
```

File: app/Http/Requests/UpdateFormRequest.php (hunk 6 above)
Comment (nicocadq) [inline app/Http/Requests/UpdateFormRequest.php:45]:
```suggestion
            'image.image' => 'No se pudo subir la imagen',
            'image.max' => 'El tamaño maximo de la imagen puede ser de hasta 1Mb',
```

**Review by nicocadq — APPROVED** [review]

File: app/Http/Requests/UpdateFormRequest.php (hunk 6 above)
Comment (alejandroGonGon) [reply app/Http/Requests/UpdateFormRequest.php]: Done

File: app/Http/Requests/UpdateFormRequest.php (hunk 4 above)
Comment (alejandroGonGon) [reply app/Http/Requests/UpdateFormRequest.php]: Done

File: app/Http/Requests/StoreDonationFormRequest.php (hunk 3 above)
Comment (alejandroGonGon) [reply app/Http/Requests/StoreDonationFormRequest.php]: Done

File: app/Http/Requests/StoreDonationFormRequest.php (hunk 2 above)
Comment (alejandroGonGon) [reply app/Http/Requests/StoreDonationFormRequest.php]: Done

File: app/Http/Requests/DonationRequestFormRequest.php [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+        return [
+            'reason' => 'required|string',
+        ];
+    }
+    
+    public function messages()
+    {
+        return [
+            'reason.required' => 'La Razon es requerida',
+            'password.string' => 'Ingrese una Razon con caracteres validos',
```
Comment (rubenderigo) [inline app/Http/Requests/DonationRequestFormRequest.php:35]:
```suggestion
            'password.string' => 'Ingrese una razón con caracteres válidos',
```

File: app/Http/Requests/PetitionStoreFormRequest.php [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+        return [
+            'subject' => 'required',
+            'description' => 'required',
+        ];
+    }
+    
+    public function messages()
+    {
+        return [
+            'subject.required' => 'El Titulo es requerido',
```
Comment (rubenderigo) [inline app/Http/Requests/PetitionStoreFormRequest.php:35]:
```suggestion
            'subject.required' => 'El titulo es requerido',
```

File: app/Http/Requests/StoreDonationFormRequest.php [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+            'name' => 'required|string',
+            'description' => 'required|string',
+            'image' => 'required|mimes:jpeg,bmp,png|max:1000',
+        ];
+    }
+
+    public function messages()
+    {
+        return [
+            'name.required' => 'El Titulo es requerido',
```
Comment (rubenderigo) [inline app/Http/Requests/StoreDonationFormRequest.php:36]:
```suggestion
            'name.required' => 'El titulo es requerido',
```

File: app/Http/Requests/StoreDonationFormRequest.php [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+            'description.required' => 'La Descripcion es requerida',
```
Comment (rubenderigo) [inline app/Http/Requests/StoreDonationFormRequest.php:37]:
```suggestion
            'description.required' => 'La descripción es requerida',
```

File: app/Http/Requests/StoreDonationFormRequest.php [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+            'image.required' => 'Una imagen es requerida',
+            'image.image' => 'Se necesita una Imagen valida',
```
Comment (rubenderigo) [inline app/Http/Requests/StoreDonationFormRequest.php:39]:
```suggestion
            'image.image' => 'Se necesita una imagen valida',
```

File: app/Http/Requests/StoreDonationFormRequest.php [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+            'image.max' => 'El tamaño maximo de la imagen puede ser de hasta 5Mb',
```
Comment (rubenderigo) [inline app/Http/Requests/StoreDonationFormRequest.php:40]:
```suggestion
            'image.max' => 'El tamaño máximo de la imagen puede ser de hasta 5Mb',
```

File: app/Http/Requests/UpdateFormRequest.php [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+        ];
+    }
+    public function messages()
+    {
+        return [
+            'email.required' => 'El email es requerido',
+            'password.required' => 'La contraseña es requerida',
+            'name.required' => 'El nombre es requerido',
+            'name.string' => 'Indique un nombre valido',
+            'phone.numeric' => 'Porfavor indique un número',
```
Comment (rubenderigo) [inline app/Http/Requests/UpdateFormRequest.php:41]:
```suggestion
            'phone.numeric' => 'Por favor indique un número',
```

File: app/Http/Requests/UpdateFormRequest.php [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+            'about_me.string' => 'Porfavor indique solo texto',
```
Comment (rubenderigo) [inline app/Http/Requests/UpdateFormRequest.php:42]:
```suggestion
            'about_me.string' => 'Por favor indique solo texto',
```

File: app/Http/Requests/UpdateFormRequest.php [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+            'image.image' => 'No se pudo subir la imagen',
+            'image.max' => 'El tamaño maximo de la imagen puede ser de hasta 1Mb',
```
Comment (rubenderigo) [inline app/Http/Requests/UpdateFormRequest.php:44]:
```suggestion
            'image.max' => 'El tamaño máximo de la imagen puede ser de hasta 1Mb',
```

**Review by rubenderigo — CHANGES_REQUESTED** [review]

**Pull request merged**